packages/forms-web-app/tests/unit/validators/custom/date-input.test.js

```diff
@@ -0,0 +1,357 @@
+const { validationResult } = require('express-validator');
+const dateInputValidation = require('../../../../src/validators/custom/date-input');
+const { testExpressValidatorMiddleware } = require('../validation-middleware-helper');
+
+describe('validators/custom/date-input', () => {
+  const mockId = 'mock-date';
+  const mockLabel = 'the mock date';
+
+  describe('rules', () => {
+    it(`has a rule for empty day input`, () => {
+      const rule = dateInputValidation(mockId, mockLabel)[0].builder.build();
+
+      expect(rule.fields).toEqual(['mock-date-day']);
+      expect(rule.locations).toEqual(['body']);
+      expect(rule.optional).toBeFalsy();
+      expect(rule.stack[0].validator.name).toEqual('isEmpty');
+      expect(rule.stack[0].negated).toBeTruthy();
+    });
+
+    it(`has a rule for empty month input`, () => {
+      const rule = dateInputValidation(mockId, mockLabel)[1].builder.build();
+
+      expect(rule.fields).toEqual(['mock-date-month']);
+      expect(rule.locations).toEqual(['body']);
+      expect(rule.optional).toBeFalsy();
+      expect(rule.stack[0].validator.name).toEqual('isEmpty');
+      expect(rule.stack[0].negated).toBeTruthy();
+    });
+
+    it(`has a rule for empty year input`, () => {
+      const rule = dateInputValidation(mockId, mockLabel)[2].builder.build();
+
+      expect(rule.fields).toEqual(['mock-date-year']);
+      expect(rule.locations).toEqual(['body']);
+      expect(rule.optional).toBeFalsy();
+      expect(rule.stack[0].validator.name).toEqual('isEmpty');
+      expect(rule.stack[0].negated).toBeTruthy();
+    });
+
+    it(`has a rule for empty day input`, () => {
+      const rule = dateInputValidation(mockId, mockLabel)[3].builder.build();
+
+      expect(rule.fields).toEqual(['mock-date-day']);
+      expect(rule.locations).toEqual(['body']);
+      expect(rule.optional).toBeFalsy();
+      expect(rule.stack[0].options).toEqual([{ min: 1, max: 31 }]);
+      expect(rule.stack[0].negated).toBeFalsy();
+      expect(rule.stack[3].negated).toBeFalsy();
+    });
+
+    it(`has a rule for empty day input`, () => {
+      const rule = dateInputValidation(mockId, mockLabel)[4].builder.build();
+
+      expect(rule.fields).toEqual(['mock-date-month']);
+      expect(rule.locations).toEqual(['body']);
+      expect(rule.optional).toBeFalsy();
+      expect(rule.stack[0].options).toEqual([{ min: 1, max: 12 }]);
+      expect(rule.stack[0].negated).toBeFalsy();
+    });
+
+    it(`has a rule for empty day input`, () => {
```

Comment:
test name is confusing - same for lines 51, and 40

Reply:
rest looks good to me 👍

Reply:
Unfortunately there are 2 sets of rules for each input so this seemed the best way to split them. First deals with if they are empty, second deals with if they are valid